beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -241,6 +241,13 @@ func (s *Service) sendBatchRootRequest(ctx context.Context, roots [][32]byte, ra
 	ctx, span := trace.StartSpan(ctx, "sendBatchRootRequest")
 	defer span.End()
 
+	roots = dedupRoots(roots)
+	for i, root := range roots {
+		if s.cfg.chain.BlockBeingSynced(root) {
+			roots = append(roots[:i], roots[i+1:]...)
+			break
```

Comment:
Why is there a break here? `currentlySyncingBlock` can have multiple blocks. Without breaking we will run into an issue of modifying the slice while iterating over it, but iterating backwards should fix it.

Reply:
that's right, I thought I would rather avoid the loop cause being stuck in multiple blocks at the same time seems quite an edge case. Will continue the loop here, shouldn't make much difference